### ISSUE TITLE
"No songs have been changed" when adding play counts and ratings

### ISSUE THREAD

**Issue by annejohnson** [phauer/migrate-itunes-to-rhythmbox#5]
Thank you for this great script! It made it quick and easy to import my iTunes playlists into Rhythmbox. :smile: 

I'm having some trouble when trying to add play counts and ratings and was wondering if you had suggestions. Here is what I'm running.

```
migrate-itunes-to-rhythmbox counts-ratings \
    --itunes_library_xml="~/Music/iTunes/iTunes Music Library.xml" \
    --rhythmdb="~/.local/share/rhythmbox/rhythmdb.xml" \
    --source_library_root="/Users/Anne/Music/iTunes/iTunes%20Music/" \
    --target_library_root="/home/anne/Music/iTunes/iTunes%20Music/"
```

Here's the output I get:

```
Reading iTunes library in /home/anne/Music/iTunes/iTunes Music Library.xml
Adding play counts and ratings to Rhythmbox DB in /home/anne/.local/share/rhythmbox/rhythmdb.xml
No songs have been changed. Probably because the iTunes and Rhythmbox song paths could not be matched together. Please check your source_library_root and target_library_root.
```

In Rhythmbox, I see that no ratings or playcounts are imported. But based on my `iTunes Music Library.xml`, I expect many to be imported.

Do you have any suggestions on what I'm doing wrong or how to troubleshoot? 

---

If it's helpful, here is what I ran successfully to import my playlists:

```
migrate-itunes-to-rhythmbox playlists \
    --itunes_library_xml="~/Music/iTunes/iTunes Music Library.xml" \
    --rhythmbox_playlists_xml="~/.local/share/rhythmbox/playlists.xml" \
    --source_library_root="/Users/Anne/Music/iTunes/iTunes%20Music/" \
    --target_library_root="/home/anne/Music/iTunes/iTunes%20Music/"
```

Output:

```
Reading iTunes library in /home/anne/Music/iTunes/iTunes Music Library.xml
Creating Rhythmbox playlist in /home/anne/.local/share/rhythmbox/playlists.xml
Finished
```

**Comment by phauer** [comment]
Hi Anne,
hard to say. I need more information. Can you please give me some examples for song paths in your iTunes Music Library.xml, your rhythmdb.xml and your playlists.xml?
Are you migrating from a Mac?

Cheers, Philipp

**Comment by annejohnson** [comment]
Hi Philipp, thank you very much for your help!

Yes, I am migrating from a Mac. Below are some example paths from each file. Let me know if any other info would be helpful.

---

Examples from iTunes Music Library.xml: 
- `file:///Users/Anne/Music/iTunes/iTunes%20Music/The%20Beastie%20Boys/To%20The%205%20Boroughs/01%20Ch-Check%20It%20Out.mp3`
- `file:///Users/Anne/Music/iTunes/iTunes%20Music/Yeah%20Yeah%20Yeahs/Gold%20Lion%20%5BSingle%5D/02%20Let%20Me%20Know%20%5BDemo%20Version%5D.MP3`

A full item from that file:

```
<key>2050</key>
<dict>
  <key>Track ID</key><integer>2050</integer>
  <key>Size</key><integer>6391309</integer>
  <key>Total Time</key><integer>266213</integer>
  <key>Track Number</key><integer>1</integer>
  <key>Date Modified</key><date>2006-03-04T16:02:50Z</date>
  <key>Date Added</key><date>2009-07-13T00:25:51Z</date>
  <key>Bit Rate</key><integer>192</integer>
  <key>Sample Rate</key><integer>44100</integer>
  <key>Play Count</key><integer>36</integer>
  <key>Play Date</key><integer>3547037864</integer>
  <key>Play Date UTC</key><date>2016-05-25T20:17:44Z</date>
  <key>Skip Count</key><integer>6</integer>
  <key>Skip Date</key><date>2014-02-21T19:14:09Z</date>
  <key>Rating</key><integer>100</integer>
  <key>Album Rating</key><integer>80</integer>
  <key>Album Rating Computed</key><true/>
  <key>Persistent ID</key><string>7CF77B2004D88CA4</string>
  <key>Track Type</key><string>File</string>
  <key>File Folder Count</key><integer>4</integer>
  <key>Library Folder Count</key><integer>1</integer>
  <key>Name</key><string>Mama Kin</string>
  <key>Artist</key><string>Aerosmith</string>
  <key>Album</key><string>O, Yeah! Ultimate Aerosmith Hits (Disc 1)</string>
  <key>Genre</key><string>Hard Rock</string>
  <key>Kind</key><string>MPEG audio file</string>
  <key>Location</key><string>file:///Users/Anne/Music/iTunes/iTunes%20Music/Aerosmith/O,%20Yeah!%20Ultimate%20Aerosmith%20Hits%20(Disc%201)/01%20Mama%20Kin.mp3</string>
</dict>
```

---

Examples from rhythmdb.xml:
- `file:///home/anne/Music/iTunes/iTunes%20Music/The%20Beastie%20Boys/To%20The%205%20Boroughs/01%20Ch-Check%20It%20Out.mp3`
- `file:///home/anne/Music/iTunes/iTunes%20Music/Yeah%20Yeah%20Yeahs/Gold%20Lion%20%5BSingle%5D/02%20Let%20Me%20Know%20%5BDemo%20Version%5D.MP3`

Full item from that file:

```
<entry type="song">
  <title>Mama Kin</title>
  <genre>Hard Rock</genre>
  <artist>Aerosmith</artist>
  <album>O, Yeah! Ultimate Aerosmith Hits (Disc 1)</album>
  <track-number>1</track-number>
  <duration>266</duration>
  <file-size>6391309</file-size>
  <location>file:///home/anne/Music/iTunes/iTunes%20Music/Aerosmith/O,%20Yeah!%20Ultimate%20Aerosmith%20Hits%20(Disc%201)/01%20Mama%20Kin.mp3</location>
  <mtime>1141488170</mtime>
  <first-seen>1479865234</first-seen>
  <last-seen>1479867287</last-seen>
  <bitrate>191</bitrate>
  <date>0</date>
  <media-type>audio/mpeg</media-type>
  <composer>Unknown</composer>
</entry>
```

---

Examples from playlists.xml:

- `file:///home/anne/Music/iTunes/iTunes%20Music/Deadmau5%20&amp;%20Wolfgang%20Gartner/Unknown%20Album/Animal%20Rights.mp3`
- `file:///home/anne/Music/iTunes/iTunes%20Music/Aerosmith/O,%20Yeah!%20Ultimate%20Aerosmith%20Hits%20(Disc%201)/01%20Mama%20Kin.mp3`

**Comment by phauer** [comment]
Thanks!
The song paths in the itunes library.xml on a Mac seems to be different than those on Windows (there is no 'localhost' in the path). I don't use a Mac so I haven't noticed this issue yet. I'll fix this. Just give me some time.

**Comment by phauer** [comment]
Hi Anne,
I just released version 1.0.3 which should fix this issue. Can you please uninstall 1.0.2, install 1.0.3, test it and give me a short feedback?
Cheers, Philipp

**Comment by annejohnson** [comment]
Hi Philipp, it worked perfectly! :+1: I have all the expected play counts and ratings now. Thank you for addressing this!

Here is what I ran, and the output:

```
$ migrate-itunes-to-rhythmbox counts-ratings \
    --itunes_library_xml="~/Music/iTunes/iTunes Music Library.xml" \
    --rhythmdb="~/.local/share/rhythmbox/rhythmdb.xml" \
    --source_library_root="/Users/Anne/Music/iTunes/iTunes%20Music/" \
    --target_library_root="/home/anne/Music/iTunes/iTunes%20Music/"

Reading iTunes library in /home/anne/Music/iTunes/iTunes Music Library.xml
Adding play counts and ratings to Rhythmbox DB in /home/anne/.local/share/rhythmbox/rhythmdb.xml
   The itunes location http://deimos3.apple.com/WebObjects/Core.woa/Feed/new.duke.edu.2647936925.02647936930 doesn't start with a known prefix.
    It's likely that we can't match it later to a rhythmbox path.
Updated 1384 song entries
```

Thanks again! I'm so happy to have all my music data imported. Hope you have a great rest of the weekend!
